File: code/platform/src/main/kotlin/org/fbme/ide/platform/CustomFacetFactory.java
package org.fbme.ide.platform;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.mps.openapi.module.FacetsFacade;
import org.jetbrains.mps.openapi.module.SModule;
import org.jetbrains.mps.openapi.module.SModuleFacet;

public class CustomFacetFactory implements FacetsFacade.FacetFactory {
    @Override
    public boolean isApplicable(@NotNull SModule module) {
        return FacetsFacade.FacetFactory.super.isApplicable(module);
    }

    @Override
    public SModuleFacet create(@NotNull SModule sModule) {
        return new CustomFacet(sModule);
    }

    @NotNull
    @Override
    public String getPresentation() {
        return "Library";
    }

    public static CustomFacetFactory CUSTOM_FACET_FACTORY = new CustomFacetFactory();
}

File: code/platform/src/main/kotlin/org/fbme/ide/platform/CustomFacetFactory.kt
package org.fbme.ide.platform

import org.jetbrains.mps.openapi.module.FacetsFacade.FacetFactory
import org.jetbrains.mps.openapi.module.SModule
import org.jetbrains.mps.openapi.module.SModuleFacet

class CustomFacetFactory : FacetFactory {
    override fun isApplicable(module: SModule): Boolean {
        return super.isApplicable(module)
    }

    override fun create(sModule: SModule): SModuleFacet {
        return CustomFacet(sModule)
    }

    override fun getPresentation(): String {
        return "Library"
    }

    companion object {
        var CUSTOM_FACET_FACTORY = CustomFacetFactory()
    }
}
